lib/config-generator.js

```diff
@@ -115,73 +78,112 @@ export class ConfigGenerator {
-        }
+                exportContent += `  {languageOptions: { ${envContent[this.answers.env.join(",")]} }},\n`;
+            }
+
+            if (purpose === "syntax") {
+
+                // no need to install any plugin
+            } else if (purpose === "problems") {
+                this.result.devDependencies.push("@eslint/js");
+                importContent += "import pluginJs from \"@eslint/js\";\n";
+                exportContent += "  pluginJs.configs.recommended,\n";
+            }
+
+            if (useTs) {
+                extensions.push("ts");
+                this.result.devDependencies.push("typescript-eslint");
+                importContent += "import tseslint from \"typescript-eslint\";\n";
+                exportContent += "  ...tseslint.configs.recommended,\n";
+            }
+
+            if (this.answers.framework === "vue") {
+                extensions.push("vue");
+                this.result.devDependencies.push("eslint-plugin-vue");
+                importContent += "import pluginVue from \"eslint-plugin-vue\";\n";
+                exportContent += "  ...pluginVue.configs[\"flat/essential\"],\n";
+
+                // https://eslint.vuejs.org/user-guide/#how-to-use-a-custom-parser
+                if (useTs) {
+                    exportContent += "  {files: [\"**/*.vue\"], languageOptions: {parserOptions: {parser: tseslint.parser}}},\n";
+                }
+            }
+
+            if (this.answers.framework === "react") {
+                extensions.push("jsx");
 
-        if (this.answers.purpose === "syntax") {
+                if (useTs) {
+                    extensions.push("tsx");
+                }
 
-            // no need to install any plugin
-        } else if (this.answers.purpose === "problems") {
-            this.result.devDependencies.push("@eslint/js");
-            importContent += "import pluginJs from \"@eslint/js\";\n";
-            exportContent += "  pluginJs.configs.recommended,\n";
+                this.result.devDependencies.push("eslint-plugin-react");
+                importContent += "import pluginReact from \"eslint-plugin-react\";\n";
+                exportContent += "  pluginReact.configs.flat.recommended,\n";
+            }
         }
 
-        if (this.answers.language === "typescript") {
-            extensions.push("ts");
-            this.result.devDependencies.push("typescript-eslint");
-            importContent += "import tseslint from \"typescript-eslint\";\n";
-            exportContent += "  ...tseslint.configs.recommended,\n";
+        if (languages?.some(item => item.startsWith("json"))) {
+            this.result.devDependencies.push("@eslint/json");
+            importContent += "import json from \"@eslint/json\";\n";
         }
+        if (languages?.includes("json")) {
+            const config = purpose === "syntax"
+                ? "  {files: [\"**/*.json\"], language: \"json/json\"},\n"
```

Comment:
We should add `plugins: { json }` when the recommended config is not used.

(the same for `jsonc`, `json5` and `markdown`).

Reply:
[fix: add plugins](https://github.com/eslint/create-config/pull/151/commits/16e8eab8fceca0c705355130acaddf067f45a560)